File: backend/api/projects.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional
from backend.services.project_manager import get_project_manager
from backend.services.meeting_logger import MeetingLogger
from pathlib import Path

router = APIRouter()


class ProjectCreate(BaseModel):
    name: str
    description: str = ""
    genre: str = ""
    target_platform: str = ""
    driving_mode: str = "plot_driven"


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    genre: Optional[str] = None
    target_platform: Optional[str] = None
    driving_mode: Optional[str] = None


@router.post("/")
async def create_project(data: ProjectCreate):
    pm = get_project_manager()
    project = pm.create_project(
        name=data.name,
        description=data.description,
        genre=data.genre,
        target_platform=data.target_platform,
        driving_mode=data.driving_mode
    )
    return {"id": project.id, "config": project.config.model_dump()}


@router.get("/")
async def list_projects():
    pm = get_project_manager()
    projects = pm.list_projects()
    return [
        {"id": p.id, "config": p.config.model_dump() if p.config else None}
        for p in projects
    ]


@router.get("/{project_id}")
async def get_project(project_id: str):
    pm = get_project_manager()
    project = pm.get_project(project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    return {"id": project.id, "config": project.config.model_dump() if project.config else None}


@router.put("/{project_id}/config")
async def update_project_config(project_id: str, data: ProjectUpdate):
    pm = get_project_manager()
    project = pm.get_project(project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    
    update_data = {k: v for k, v in data.model_dump().items() if v is not None}
    project.update_config(**update_data)
    return {"id": project.id, "config": project.config.model_dump()}


@router.delete("/{project_id}")
async def delete_project(project_id: str):
    pm = get_project_manager()
    if pm.delete_project(project_id):
        return {"status": "deleted"}
    raise HTTPException(status_code=404, detail="Project not found")


@router.get("/{project_id}/logs")
async def get_project_logs(project_id: str, limit: int = 100):
    pm = get_project_manager()
    project = pm.get_project(project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    
    db_path = project.project_path / "logs" / "meeting.db"
    if not db_path.exists():
        return {"meeting_logs": [], "version_history": []}
    
    logger = MeetingLogger(db_path)
    return {
        "meeting_logs": logger.get_meeting_logs(project_id, limit),
        "version_history": logger.get_version_history(project_id)
    }

File: backend-node/src/routes/projects.ts
// 项目管理路由
import { Express, Request, Response } from 'express';
import { getProjectManager } from '../services/project-manager';

export function registerProjectRoutes(app: Express): void {
  const pm = getProjectManager();

  // 列出所有项目
  app.get('/api/projects', (req: Request, res: Response) => {
    const projects = pm.listProjects();
    res.json(projects.map(p => ({
      id: p.id,
      config: p.config
    })));
  });

  // 创建项目
  app.post('/api/projects', (req: Request, res: Response) => {
    const { name, description, genre, targetPlatform, drivingMode } = req.body;
    
    if (!name) {
      return res.status(400).json({ error: 'Name is required' });
    }

    const project = pm.createProject(name, {
      description,
      genre,
      targetPlatform,
      drivingMode
    });

    res.json({ id: project.id, config: project.config });
  });

  // 获取项目详情
  app.get('/api/projects/:projectId', (req: Request, res: Response) => {
    const project = pm.getProject(req.params.projectId);
    if (!project) {
      return res.status(404).json({ error: 'Project not found' });
    }
    res.json({ id: project.id, config: project.config });
  });

  // 更新项目
  app.put('/api/projects/:projectId', (req: Request, res: Response) => {
    const project = pm.updateProject(req.params.projectId, req.body);
    if (!project) {
      return res.status(404).json({ error: 'Project not found' });
    }
    res.json({ id: project.id, config: project.config });
  });

  // 删除项目
  app.delete('/api/projects/:projectId', (req: Request, res: Response) => {
    const success = pm.deleteProject(req.params.projectId);
    if (!success) {
      return res.status(404).json({ error: 'Project not found' });
    }
    res.json({ status: 'deleted' });
  });
}
